redash/tasks/queries.py

```diff
@@ -366,10 +374,7 @@ def run(self):
             result = QueryExecutionError(error)
             if self.scheduled_query is not None:
                 self.scheduled_query = models.db.session.merge(self.scheduled_query, load=False)
-                self.scheduled_query.schedule_failures += 1
-                models.db.session.add(self.scheduled_query)
-                notify_of_failure(error, self.scheduled_query)
-            models.db.session.commit()
+                track_failure(self.scheduled_query, error)
```

Comment:
Previously, `models.db.session.commit` was happening in all cases. Now it happens only on failure.

Reply:
First - I think you mean it was happening for all queries and now it only happens with scheduled queries?

Second - I have a hunch this was a mistake with indentation, as there is no `db.session` work prior to it.

Reply:
I thought it's for saving the result, but that actually happens in a different branch of the code. So I guess you're right.